src/rust/lib/frp/src/nodes.rs

```diff
@@ -0,0 +1,1229 @@
+        self.register(Map2::new(label,t1,t2,f))
+    }}
+
+    docs_for_map! {
+    pub fn map3<T1,T2,T3,F,T>
+    (&self, label:Label, t1:&T1, t2:&T2, t3:&T3, f:F) -> Stream<T>
+    where T1:EventOutput, T2:EventOutput, T3:EventOutput, T:Data,
+          F:'static+Fn(&Output<T1>,&Output<T2>,&Output<T3>)->T {
+        self.register(Map3::new(label,t1,t2,t3,f))
+    }}
+
+    docs_for_map! {
+    pub fn map4<T1,T2,T3,T4,F,T>
+    (&self, label:Label, t1:&T1, t2:&T2, t3:&T3, t4:&T4, f:F) -> Stream<T>
+    where T1:EventOutput, T2:EventOutput, T3:EventOutput, T4:EventOutput, T:Data,
+          F:'static+Fn(&Output<T1>,&Output<T2>,&Output<T3>,&Output<T4>)->T {
+        self.register(Map4::new(label,t1,t2,t3,t4,f))
+    }}
+
+
+    // === Apply ===
+
+    docs_for_apply! {
+    pub fn apply2<T1,T2,F,T>(&self, label:Label, t1:&T1, t2:&T2, f:F) -> Stream<T>
+    where T1:EventOutput, T2:EventOutput, T:Data, F:'static+Fn(&Output<T1>,&Output<T2>)->T {
+        self.register(Apply2::new(label,t1,t2,f))
+    }}
+
+    docs_for_apply! {
+    pub fn apply3<T1,T2,T3,F,T>
+    (&self, label:Label, t1:&T1, t2:&T2, t3:&T3, f:F) -> Stream<T>
+    where T1:EventOutput, T2:EventOutput, T3:EventOutput, T:Data,
+          F:'static+Fn(&Output<T1>,&Output<T2>,&Output<T3>)->T {
+        self.register(Apply3::new(label,t1,t2,t3,f))
+    }}
+
+    docs_for_apply! {
+    pub fn apply4<T1,T2,T3,T4,F,T>
+    (&self, label:Label, t1:&T1, t2:&T2, t3:&T3, t4:&T4, f:F) -> Stream<T>
+    where T1:EventOutput, T2:EventOutput, T3:EventOutput, T4:EventOutput, T:Data,
+          F:'static+Fn(&Output<T1>,&Output<T2>,&Output<T3>,&Output<T4>)->T {
+        self.register(Apply4::new(label,t1,t2,t3,t4,f))
+    }}
+}
+
+
+
+///////////////////////////////////
+//
+//
+//
+//#[allow(unused_variables)]
+//pub fn test() {
+//    println!("hello");
+//
+////    new_network! { network
+////        def source  = source::<f32>();
+////        def source2 = source::<()>();
+////        def tg      = toggle(&source);
+////        def fff     = map(&tg,|t| { println!("{:?}",t) });
+////        def bb      = sample(&source2,&tg);
+////
+////        let bb2 : Stream<bool> = bb.into();
+////
+////        def fff2   = map(&bb2,|t| { println!(">> {:?}",t) });
+////        def m      = merge_::<usize>();
+////        def c      = count(&m);
+////        def t      = trace("t",&c);
+////    }
+////
+////    m.add(&c);
+////
+////    println!("{:?}",tg);
+////
+////    source.emit(&5.0);
+////    source2.emit(&());
+////    source.emit(&5.0);
+////    source2.emit(&());
+////    source.emit(&5.0);
+////
+////    m.emit(&0);
+////    m.emit(&0);
+////    m.emit(&0);
+//
+////    network.draw();
+//
+//    crate::new_network! { network1
+//        def source = source();
+//        def count  = source.count();
+//        def t      = trace("source",&source);
+//        def t2     = trace("count",&count);
+//    }
+//
+//    source.emit(());
+//    source.emit(());
+//    source.emit(());
+//
+//}
```

Comment:
Commented code

Maybe we should have a crate like frp_tests with many binaries which draws the frp?